packages/uxpin-merge-cli/src/steps/serialization/component/implementation/typescript/component/isDefaultExported.ts

```diff
@@ -1,11 +1,103 @@
 import * as ts from 'typescript';
+import { hasUXPinComponentComment } from '../comments/hasUXPinComponentComment';
+import { TSSerializationContext } from '../context/getSerializationContext';
+import { getNodeName } from '../node/getNodeName';
+import { getComponentName } from './getComponentName';
 import { ComponentDeclaration } from './getPropsTypeAndDefaultProps';
 import { isExported } from './isExported';
 
-export function isDefaultExported(node:ComponentDeclaration):boolean {
-  if (!node.modifiers) {
+export function isDefaultExported(declaration:ComponentDeclaration, context:TSSerializationContext):boolean {
+  const componentName:string = getComponentName(context, declaration);
+
+  // e.g. export default () => {};
+  if (ts.isArrowFunction(declaration) && ts.isExportAssignment(declaration.parent)) {
+    return true;
+  }
+
+  // e.g. export default Class Component
+  // e.g. export default function Component()
+  // NOTE: This returns `false` for cases like below. So, it is too early to tell
+  // if this file export Component as default or not.
+  // And it will be checked with `isWrappedWithHOC` at later steps
+  // ------------------- Case #1
+  // /**
+  //  * @uxpincomponent
+  //  */
+  // export class Button
+  // export default withHOC(Component)
+  // ------------------- Case #2
+  // export function FunctionMatchWithFileName()| export class ClassMatchWithFileName
+  // export default withHOC(FunctionMatchWithFileName|ClassMatchWithFileName)
+  if (!hasUXPinComponentComment(declaration) && hasDefaultKeywordInModifiers(declaration)) {
+    return true;
+  }
+
+  // export default Component;
+  // Because export statement is in different node, we need to go through again...
+  let isDefault:boolean = false;
+  ts.forEachChild(context.file, (node) => {
+    if (!ts.isExportAssignment(node)) { return; }
```

Comment:
I think you didnt handle export { Component as default } because it will be ExportDeclaration

Reply:
Great catch! I've added another logic to handle this case.